qiime2/plugin/plugin.py

```diff
@@ -327,11 +613,80 @@ def __init__(self, plugin):
 
 
 class PluginMethods(PluginActions):
+    """Accessed via ``plugin.methods``"""
     def register_function(self, function, inputs, parameters, outputs, name,
                           description, input_descriptions=None,
                           parameter_descriptions=None,
                           output_descriptions=None, citations=None,
                           deprecated=False, examples=None):
+        """Register a method to the associated plugin.
+
+        Parameters
+        ----------
+        function : callable
+          A function which will be called when the user uses this method.
+        inputs : dict[str, type expression]
+          A dictionary of function-parameter names to semantic type
+          expressions. The keys of the dictionary must match the parameter
+          names which are on the ``function``. Collections and type variables
+          are also permitted so long as they contain semantic types.
+        parameters : dict[str, type expression]
+          A dictionary of function0parameter names to primitive type
```

Comment:
```suggestion
          A dictionary of function parameter names to primitive type
```

Reply:
lol oops. You can tell I was "done." with all of the typing and then copy-pasted.